site/docs/spark-queries.md

```diff
@@ -219,9 +221,33 @@ SELECT * FROM prod.db.table.files
 +-------------------------------------------------------------------------+-------------+--------------+--------------------+--------------------+------------------+-------------------+------------------+-----------------+-----------------+--------------+---------------+
 ```
 
+### All Data Files
+
+To show a table's valid data files and each file's metadata, run:
+
+!!! Note
+    a valid data file is one that is readable from any snapshot currently tracked by the table.
+
+!!! WARNING
+    this table may return **duplicate** rows
+
+```sql
+SELECT * FROM prod.db.table.all_data_files
+```
+
+```text
++-------+------------------------------+-----------+----------+------------+------------------+------------------+------------------+------------------+----------------+-----------------------+-----------------------+------------+-------------+------------+-------------+
+|content|                     file_path|file_format| partition|record_count|file_size_in_bytes|      column_sizes|      value_counts| null_value_counts|nan_value_counts|           lower_bounds|           upper_bounds|key_metadata|split_offsets|equality_ids|sort_order_id|
++-------+------------------------------+-----------+----------+------------+------------------+------------------+------------------+------------------+----------------+-----------------------+-----------------------+------------+-------------+------------+-------------+
+|      0|s3:.../dt=20210102/xxx.parquet|    PARQUET|{20210102}|          14|              2444|{1 -> 94, 2 -> 17}|{1 -> 14, 2 -> 14}|  {1 -> 0, 2 -> 0}|              {}|{1 -> 1, 2 -> 20210102}|{1 -> 2, 2 -> 20210102}|        null|          [4]|        null|            0|
+|      0|s3:.../dt=20210103/xxx.parquet|    PARQUET|{20210103}|          14|              2444|{1 -> 94, 2 -> 17}|{1 -> 14, 2 -> 14}|  {1 -> 0, 2 -> 0}|              {}|{1 -> 1, 2 -> 20210103}|{1 -> 3, 2 -> 20210103}|        null|          [4]|        null|            0|
+|      0|s3:.../dt=20210104/xxx.parquet|    PARQUET|{20210104}|          14|              2444|{1 -> 94, 2 -> 17}|{1 -> 14, 2 -> 14}|  {1 -> 0, 2 -> 0}|              {}|{1 -> 1, 2 -> 20210104}|{1 -> 3, 2 -> 20210104}|        null|          [4]|        null|            0|
++-------+------------------------------+-----------+----------+------------+------------------+------------------+------------------+------------------+----------------+-----------------------+-----------------------+------------+-------------+------------+-------------+
```

Comment:
Can you fix the formatting of this table? You can see my recommendations on how to get a nicely formatted table here: https://github.com/apache/iceberg/pull/3422#discussion_r739848024